### PR TITLE
feat: Optimize quota checking when pushing images

### PR DESCRIPTION
**Pull request by lengrongfu**
Thank you for contributing to Harbor!

# Comprehensive Summary of your change
Create a zero-length resource when calling the `post /v2/library/ubuntu/blobs/uploads/` API to check if the quota is full.

# Issue being fixed
Fixes https://github.com/goharbor/harbor/issues/17390

Please indicate you've done the following:
- [x] Well Written Title and Summary of the PR
- [x] Label the PR as needed. "release-note/new-feature,"
- [X] Accepted the DCO. Commits without the DCO will delay acceptance.
- [X] Made sure tests are passing and test coverage is added if needed.
- [x] Considered the docs impact and opened a new docs issue or PR with docs changes if needed in [website repository](https://github.com/goharbor/website).

### CODE REVIEW

**Comment by OrlinVasilev** [comment]
@lengrongfu thank you for your PR super happy to see your contribution! someone from the @goharbor/all-maintainers  will review soon :)

**Comment by wy65701436** [comment]
@lengrongfu I will take the time to evaluate this PR, thank you for your contribution.

**Review by Vad1mo — APPROVED** [review]

**Comment by lengrongfu** [comment]
@OrlinVasilev Hi, can we arrange maintainers who have time or are willing to review.

**Comment by OrlinVasilev** [comment]
Hi @lengrongfu will add it to the next community meeting agenda for review but it will be great if you can join us there! Thanks!

**Comment by lengrongfu** [comment]
> Hi @lengrongfu will add it to the next community meeting agenda for review but it will be great if you can join us there! Thanks!

I'd love to, I'll get in touch with you on Slack, thanks.

**Comment by OrlinVasilev** [comment]
@lengrongfu can you please remind me did you presented that on the community meetings ? 
@wy65701436 @MinerYang  @AllForNothing can you please check this one

**Review by chlins — APPROVED** [review]
lgtm

**Review by wy65701436 — APPROVED** [review]
lgtm

File: src/pkg/quota/util.go [hunk 1]
```diff
@@ -32,7 +32,7 @@ func IsSafe(hardLimits types.ResourceList, currentUsed types.ResourceList, newUs
 			continue
 		}
 
-		if hardLimit == types.UNLIMITED || value == currentUsed[resource] {
```
Comment (chlins) [inline src/pkg/quota/util.go:35]: @lengrongfu You removed the condition `value == currentUsed[resource]`, could you explain the reason for this?

**Pull request merged**